src/ui/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,133 @@
+import { FC, ReactNode, useState } from 'react';
+import * as RadixTooltip from '@radix-ui/react-tooltip';
+import tw, { css } from 'twin.macro';
+import { AnimatePresence, motion, Variants } from 'framer-motion';
+
+type TooltipProps = Omit<RadixTooltip.TooltipProps, 'open' | 'onOpenChange'> & {
+  /**
+   * *A single {@link JSX.Element}* as the handle to control the tooltip popover to be rendered where {@link Tooltip} is used.
+   *
+   * **Note:** A valid trigger here must be an HTML native element or an element that implement the {@link React.forwardRef} API.
+   */
+  trigger: ReactNode;
+  /**
+   * Theme color to be applied on the tooltip popover
+   */
+  theme?: 'dark' | 'light' | 'primary' | 'secondary';
+  /**
+   * Props passed into the tooltip popover element
+   */
+  menuPopupProps?: RadixTooltip.TooltipContentProps;
+};
+
+/**
+ * A component renders a popover and a handle that shows the popover on hover and hide it on leave.
+ *
+ * @example
+ * <Tooltip trigger={<Icon icon={circleQuestionMarkFiLL} />}>
+ *  This text shows up in a popover when the user hovers on the trigger
+ * </Tooltip>
+ */
+const Tooltip: FC<TooltipProps> = ({
+  trigger,
+  theme,
+  menuPopupProps,
+  children,
+  ...props
+}) => {
+  const [showTooltip, setShowTooltip] = useState(props.defaultOpen);
+  return (
+    <RadixTooltip.Root
+      delayDuration={0}
+      {...props}
+      open={showTooltip}
+      onOpenChange={setShowTooltip}
+    >
+      <RadixTooltip.Trigger asChild>{trigger}</RadixTooltip.Trigger>
+      <AnimatePresence>
+        {/* Required to aniamte exit */}
+        {showTooltip && (
+          <RadixTooltip.Content
+            side="top"
+            sideOffset={4}
+            {...menuPopupProps}
+            asChild
+            forceMount
```

Comment:
what is `forceMount`?

Reply:
🤷 

According to Radix UI's docs:

```
forceMount?: true | undefined: Used to force mounting when more control is needed. Useful when controlling animation with React animation libraries.
```

Exit aniamtion doesn't work without this